pkg/gomod/selector.go

```diff
@@ -0,0 +1,104 @@
+package gomod
+
+import (
+	"log"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+	"knative.dev/test-infra/pkg/gowork"
+)
+
+var (
+	// ErrNoDomain is returned when no domain is provided.
+	ErrNoDomain = errors.New("no domain provided")
+)
+
+// Matcher matches a module name.
+type Matcher interface {
+	// Match returns true if the module name matches.
+	Match(modname string) bool
+}
+
+// MatcherFunc is a function that implements the Matcher interface.
+func MatcherFunc(fn matcherFunc) Matcher {
+	return fn
+}
+
+type matcherFunc func(modname string) bool
+
+func (fn matcherFunc) Match(modname string) bool {
+	return fn(modname)
+}
+
+// Selector is a set of matchers that can be used to select a subset of modules.
+type Selector struct {
+	Includes []Matcher
+	Excludes []Matcher
+}
+
+// Select returns true if the module should be selected.
+func (s Selector) Select(modname string) bool {
```

Comment:
If you rename `Select` to `Match`, `Selector` becomes a `Matcher` implementation.

(Yes, this might push against the removal of the `Matcher` interface separate from the type alias.)

Reply:
Maybe a good idea to replace `Select -> Match`. That would allow users to build composite `Matcher`'s using `Selector` as a Matcher itself. :grin: